solarkraft/src/state/value.ts

```diff
@@ -11,133 +53,125 @@ interface Value { }
  * 
  * Example: 2u32 would be represented as { val: 2, type: "u32" }, whereas 2i32 would be { val: 2, type: "i32" }
 */
-abstract class IntValue implements Value {
+export type IntValue = {
     val: bigint
-    type: string
+} & (IntValue_u32 | IntValue_i32 | IntValue_u64 | IntValue_i64 | IntValue_u128 | IntValue_i128)
```

Comment:
Why would not you simply enumerate all cases right here? You still have to define types for `IntValue_u32` below.

For instance, this is just immediate and does not require any scrolling:

```ts
export type IntValue =
    { val: bigint, type: "u32" }
  | { val: bigint, type: "i32" }
  | { val: bigint, type: "u64" }
  ...
```

Alternatively, we could do:

```ts
export type IntValue =
  { type: "uint", bits: 32 | 64 | 128, val: bigint }
  { type: "int", bits: 32 | 64 | 128, val: bigint }
```